opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1886,6 +1886,33 @@ namespace {
     }
 
     template<class T>
+    double
+    FullyImplicitBlackoilSolver<T>::convergenceReduction(const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& B,
+                                                         const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& tempV,
+                                                         const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& R,
+                                                         std::array<double,MaxNumPhases>& R_sum,
+                                                         std::array<double,MaxNumPhases>& maxCoeff,
+                                                         std::array<double,MaxNumPhases>& B_avg,
+                                                         int nc) const
+    {
+        const Opm::PhaseUsage& pu = fluid_.phaseUsage();
+        // Do the global reductions
+        for(int idx=0; idx<MaxNumPhases; ++idx)
```

Comment:
Another note on active phases: with the current code it is most convenient to output convergence numbers for all phases, active or not. That means the idea of making everything zero for inactive phases is good. I think if you write to R_sum[idx] (etc.) in this loop, things will be working as expected.